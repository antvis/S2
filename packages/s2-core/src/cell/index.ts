import { BaseCell } from './base-cell';
import { ColCell } from './col-cell';
import { CornerCell } from './corner-cell';
import { DataCell } from './data-cell';
import { MergedCell } from './merged-cell';
import { RowCell } from './row-cell';
import { TableColCell } from './table-col-cell';
import { TableCornerCell } from './table-corner-cell';
import { TableDataCell } from './table-data-cell';
import { TableRowCell } from './table-series-cell';

export {
  TableCornerCell,
  TableColCell,
  TableRowCell,
  RowCell,
  ColCell,
  DataCell,
  MergedCell,
  CornerCell,
  BaseCell,
  TableDataCell,
};
