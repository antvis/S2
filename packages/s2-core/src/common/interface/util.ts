export type DeepRequired<T extends Record<string, any>> = {
  [K in keyof T]-?: NonNullable<T[K]> extends Record<string, any>
    ? DeepRequired<NonNullable<T[K]>>
    : NonNullable<T[K]>;
};
