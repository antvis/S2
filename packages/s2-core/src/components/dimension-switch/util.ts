import { DimensionType, DimensionItem } from './dimension';

export interface OperatedType {
  [type: string]: string[];
}

export const getDimensionsByPredicate = (
  data: DimensionType[],
  predicate: (value: DimensionItem) => boolean,
) => {
  return data.reduce((obj, t) => {
    obj[t.type] = t.items.filter(predicate).map((i) => i.id);
    return obj;
  }, {} as OperatedType);
};
