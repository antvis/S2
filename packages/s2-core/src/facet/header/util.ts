import { isNil } from 'lodash';

export const getCellPadding = () => {
  const padding = [12, 4, 12, 4];
  const left = isNil(padding[3]) ? 0 : padding[3];
  const right = isNil(padding[1]) ? 0 : padding[1];
  const top = isNil(padding[0]) ? 0 : padding[0];
  const bottom = isNil(padding[2]) ? 0 : padding[2];
  return {
    left,
    right,
    top,
    bottom,
  };
};
