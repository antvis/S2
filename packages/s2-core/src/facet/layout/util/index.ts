export { default as getDimsConditionByNode } from './get-dims-condition-by-node';
export { default as getColHeight } from './get-col-height';
export { default as getColWidth } from './get-col-width';
export { default as getRowWidth } from './get-row-width';
export { default as processColsList } from './process-cols-list';
export { default as getHeaderHierarchyQuery } from './get-header-hierarchy-query';
