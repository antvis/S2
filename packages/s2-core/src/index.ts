export { Event as GEvent } from '@antv/g-canvas';
export { getTheme } from './theme';
export { Node } from './facet/layout/node';
export { Hierarchy } from './facet/layout/hierarchy';
export { buildTableHierarchy } from './facet/layout/build-table-hierarchy';
export * from './common';
export * from './utils';
export * from './cell';
export * from './sheet-type';
export * from './data-set';
export * from './interaction';
export * from './ui/tooltip';
export * from './facet/header';
