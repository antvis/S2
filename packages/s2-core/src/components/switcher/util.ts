import { filter, flatten, isEmpty, isNil, map } from 'lodash';
import { DraggableLocation } from 'react-beautiful-dnd';
import {
  FieldType,
  MAX_DIMENSION_COUNT,
  SWITCHER_PREFIX_CLS,
} from './constant';
import { SwitcherItem, SwitchResult, SwitchState } from './interface';
import { getClassNameWithPrefix } from '@/utils/get-classnames';

export const getSwitcherClassName = (...classNames: string[]) =>
  getClassNameWithPrefix(SWITCHER_PREFIX_CLS, ...classNames);

export const getNonEmptyFieldCount = (state: SwitchState) => {
  return [
    state[FieldType.Rows],
    state[FieldType.Cols],
    state[FieldType.Values],
  ].reduce((sum, value) => sum + (isEmpty(value) ? 0 : 1), 0);
};

export const getMainLayoutClassName = (nonEmptyCount: number) => {
  switch (nonEmptyCount) {
    case 1:
      return getSwitcherClassName('content', 'one-dimension');
    case 2:
      return getSwitcherClassName('content', 'two-dimensions');
    default:
      return getSwitcherClassName('content', 'three-dimensions');
  }
};

export const showDimensionCrossRows = (nonEmptyCount: number) =>
  nonEmptyCount < MAX_DIMENSION_COUNT;

export const isMeasureType = (fieldType: FieldType) =>
  fieldType === FieldType.Values;

export const moveItem = (
  source: SwitcherItem[],
  destination: SwitcherItem[],
  droppableSource: DraggableLocation,
  droppableDestination: DraggableLocation,
): SwitchState => {
  // change order in same column
  if (droppableDestination.droppableId === droppableSource.droppableId) {
    const updatingDestination = [...destination];
    const [removed] = updatingDestination.splice(droppableSource.index, 1);
    updatingDestination.splice(droppableDestination.index, 0, removed);
    return {
      [droppableDestination.droppableId]: updatingDestination,
    };
  }
  // move to other column
  const updatingSource = [...source];
  const updatingDestination = [...destination];

  const [removed] = updatingSource.splice(droppableSource.index, 1);
  updatingDestination.splice(droppableDestination.index, 0, removed);

  return {
    [droppableSource.droppableId]: updatingSource,
    [droppableDestination.droppableId]: updatingDestination,
  };
};

export const checkItem = (
  source: SwitcherItem[],
  checked: boolean,
  id: string,
  parentId?: string,
): SwitcherItem[] => {
  const target: SwitcherItem = {
    ...source.find((item) => item.id === (parentId ?? id)),
  };

  // 有 parentId 时，说明是第二层次项的改变
  if (parentId) {
    target.children = map(target.children, (item) => ({
      ...item,
      checked: item.id === id ? checked : item.checked,
    }));
  } else {
    target.checked = checked;
    target.children = map(target.children, (item) => ({
      ...item,
      checked: checked,
    }));
  }

  return source.map((item) => (item.id === target.id ? target : item));
};

export const generateSwitchResult = (state: SwitchState): SwitchResult => {
  // rows and cols can't be hidden
  const rows = map(state[FieldType.Rows], 'id');
  const cols = map(state[FieldType.Cols], 'id');

  // flatten all values and derived values
  const values = flatten(
    map(state[FieldType.Values], (item) => {
      const children = map(item.children, 'id');
      return [item.id, ...children];
    }),
  );

  const filterHiddenValues = (item: SwitcherItem) => item.checked === false;

  //  get all hidden values
  const hiddenValues = flatten(
    map(filter(state[FieldType.Values], filterHiddenValues), (item) => {
      const hiddenChildren = map(
        filter(item.children, filterHiddenValues),
        'id',
      );
      return [item.id, ...hiddenChildren];
    }),
  );

  return { rows, cols, values, hiddenValues };
};
