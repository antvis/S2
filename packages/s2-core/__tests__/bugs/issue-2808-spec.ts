/**
 * @description spec for issue #2808
 * https://github.com/antvis/S2/issues/2808
 */
import { type S2Options } from '../../src';
import { createPivotSheet } from '../util/helpers';

describe('Cell Nill Value Tests', () => {
  const getCellMetaList = async (s2Options?: S2Options) => {
    const s2 = createPivotSheet(
      { width: 800, height: 600, ...s2Options },
      { useSimpleData: false },
    );

    const data = [
      {
        number: 7789,
        province: null,
        city: null,
        type: null,
        sub_type: null,
      },
      ...s2.dataCfg.data,
    ];

    s2.setDataCfg({
      data,
    });

    await s2.render();

    return s2.facet.getCells().map((cell) => {
      const meta = cell.getMeta();

      return {
        id: meta.id,
        value: meta.value || meta.fieldValue,
        actualText: cell.getActualText(),
      };
    });
  };

  test('should get correctly nill cell value', async () => {
    const metaList = await getCellMetaList();

    expect(metaList).toMatchSnapshot();
  });

  test('should get correctly empty placeholder cell value', async () => {
    const metaList = await getCellMetaList({
      placeholder: {
        cell: '我是占位符',
      },
    });

    expect(metaList).toMatchSnapshot();
  });
});
