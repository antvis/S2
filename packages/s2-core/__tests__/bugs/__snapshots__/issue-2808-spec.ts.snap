// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Nill Value Tests should get correctly empty placeholder cell value 1`] = `
Array [
  Object {
    "actualText": "省份",
    "id": "province",
    "value": "省份",
  },
  Object {
    "actualText": "城市",
    "id": "city",
    "value": "城市",
  },
  Object {
    "actualText": "类别",
    "id": "type",
    "value": "类别",
  },
  Object {
    "actualText": "子类别",
    "id": "sub_type",
    "value": "子类别",
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "浙江省",
    "id": "root[&]浙江省",
    "value": "浙江省",
  },
  Object {
    "actualText": "杭州市",
    "id": "root[&]浙江省[&]杭州市",
    "value": "杭州市",
  },
  Object {
    "actualText": "绍兴市",
    "id": "root[&]浙江省[&]绍兴市",
    "value": "绍兴市",
  },
  Object {
    "actualText": "宁波市",
    "id": "root[&]浙江省[&]宁波市",
    "value": "宁波市",
  },
  Object {
    "actualText": "舟山市",
    "id": "root[&]浙江省[&]舟山市",
    "value": "舟山市",
  },
  Object {
    "actualText": "四川省",
    "id": "root[&]四川省",
    "value": "四川省",
  },
  Object {
    "actualText": "成都市",
    "id": "root[&]四川省[&]成都市",
    "value": "成都市",
  },
  Object {
    "actualText": "绵阳市",
    "id": "root[&]四川省[&]绵阳市",
    "value": "绵阳市",
  },
  Object {
    "actualText": "南充市",
    "id": "root[&]四川省[&]南充市",
    "value": "南充市",
  },
  Object {
    "actualText": "乐山市",
    "id": "root[&]四川省[&]乐山市",
    "value": "乐山市",
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "数量",
    "id": "root[&]$$null$$[&]$$null$$[&]number",
    "value": "number",
  },
  Object {
    "actualText": "家具",
    "id": "root[&]家具",
    "value": "家具",
  },
  Object {
    "actualText": "桌子",
    "id": "root[&]家具[&]桌子",
    "value": "桌子",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]家具[&]桌子[&]number",
    "value": "number",
  },
  Object {
    "actualText": "沙发",
    "id": "root[&]家具[&]沙发",
    "value": "沙发",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]家具[&]沙发[&]number",
    "value": "number",
  },
  Object {
    "actualText": "办公用品",
    "id": "root[&]办公用品",
    "value": "办公用品",
  },
  Object {
    "actualText": "笔",
    "id": "root[&]办公用品[&]笔",
    "value": "笔",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]办公用品[&]笔[&]number",
    "value": "number",
  },
  Object {
    "actualText": "纸张",
    "id": "root[&]办公用品[&]纸张",
    "value": "纸张",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]办公用品[&]纸张[&]number",
    "value": "number",
  },
  Object {
    "actualText": "7789",
    "id": "root[&]$$null$$[&]$$null$$-root[&]$$null$$[&]$$null$$[&]number",
    "value": 7789,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]浙江省[&]杭州市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]浙江省[&]绍兴市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]浙江省[&]宁波市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]浙江省[&]舟山市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]四川省[&]成都市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]四川省[&]绵阳市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]四川省[&]南充市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]四川省[&]乐山市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$-root[&]家具[&]桌子[&]number",
    "value": null,
  },
  Object {
    "actualText": "7789",
    "id": "root[&]浙江省[&]杭州市-root[&]家具[&]桌子[&]number",
    "value": 7789,
  },
  Object {
    "actualText": "2367",
    "id": "root[&]浙江省[&]绍兴市-root[&]家具[&]桌子[&]number",
    "value": 2367,
  },
  Object {
    "actualText": "3877",
    "id": "root[&]浙江省[&]宁波市-root[&]家具[&]桌子[&]number",
    "value": 3877,
  },
  Object {
    "actualText": "4342",
    "id": "root[&]浙江省[&]舟山市-root[&]家具[&]桌子[&]number",
    "value": 4342,
  },
  Object {
    "actualText": "1723",
    "id": "root[&]四川省[&]成都市-root[&]家具[&]桌子[&]number",
    "value": 1723,
  },
  Object {
    "actualText": "1822",
    "id": "root[&]四川省[&]绵阳市-root[&]家具[&]桌子[&]number",
    "value": 1822,
  },
  Object {
    "actualText": "1943",
    "id": "root[&]四川省[&]南充市-root[&]家具[&]桌子[&]number",
    "value": 1943,
  },
  Object {
    "actualText": "2330",
    "id": "root[&]四川省[&]乐山市-root[&]家具[&]桌子[&]number",
    "value": 2330,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$-root[&]家具[&]沙发[&]number",
    "value": null,
  },
  Object {
    "actualText": "5343",
    "id": "root[&]浙江省[&]杭州市-root[&]家具[&]沙发[&]number",
    "value": 5343,
  },
  Object {
    "actualText": "632",
    "id": "root[&]浙江省[&]绍兴市-root[&]家具[&]沙发[&]number",
    "value": 632,
  },
  Object {
    "actualText": "7234",
    "id": "root[&]浙江省[&]宁波市-root[&]家具[&]沙发[&]number",
    "value": 7234,
  },
  Object {
    "actualText": "834",
    "id": "root[&]浙江省[&]舟山市-root[&]家具[&]沙发[&]number",
    "value": 834,
  },
  Object {
    "actualText": "2451",
    "id": "root[&]四川省[&]成都市-root[&]家具[&]沙发[&]number",
    "value": 2451,
  },
  Object {
    "actualText": "2244",
    "id": "root[&]四川省[&]绵阳市-root[&]家具[&]沙发[&]number",
    "value": 2244,
  },
  Object {
    "actualText": "2333",
    "id": "root[&]四川省[&]南充市-root[&]家具[&]沙发[&]number",
    "value": 2333,
  },
  Object {
    "actualText": "2445",
    "id": "root[&]四川省[&]乐山市-root[&]家具[&]沙发[&]number",
    "value": 2445,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$-root[&]办公用品[&]笔[&]number",
    "value": null,
  },
  Object {
    "actualText": "945",
    "id": "root[&]浙江省[&]杭州市-root[&]办公用品[&]笔[&]number",
    "value": 945,
  },
  Object {
    "actualText": "1304",
    "id": "root[&]浙江省[&]绍兴市-root[&]办公用品[&]笔[&]number",
    "value": 1304,
  },
  Object {
    "actualText": "1145",
    "id": "root[&]浙江省[&]宁波市-root[&]办公用品[&]笔[&]number",
    "value": 1145,
  },
  Object {
    "actualText": "1432",
    "id": "root[&]浙江省[&]舟山市-root[&]办公用品[&]笔[&]number",
    "value": 1432,
  },
  Object {
    "actualText": "2335",
    "id": "root[&]四川省[&]成都市-root[&]办公用品[&]笔[&]number",
    "value": 2335,
  },
  Object {
    "actualText": "245",
    "id": "root[&]四川省[&]绵阳市-root[&]办公用品[&]笔[&]number",
    "value": 245,
  },
  Object {
    "actualText": "2457",
    "id": "root[&]四川省[&]南充市-root[&]办公用品[&]笔[&]number",
    "value": 2457,
  },
  Object {
    "actualText": "2458",
    "id": "root[&]四川省[&]乐山市-root[&]办公用品[&]笔[&]number",
    "value": 2458,
  },
  Object {
    "actualText": "我是占位符",
    "id": "root[&]$$null$$[&]$$null$$-root[&]办公用品[&]纸张[&]number",
    "value": null,
  },
  Object {
    "actualText": "1343",
    "id": "root[&]浙江省[&]杭州市-root[&]办公用品[&]纸张[&]number",
    "value": 1343,
  },
  Object {
    "actualText": "1354",
    "id": "root[&]浙江省[&]绍兴市-root[&]办公用品[&]纸张[&]number",
    "value": 1354,
  },
  Object {
    "actualText": "1523",
    "id": "root[&]浙江省[&]宁波市-root[&]办公用品[&]纸张[&]number",
    "value": 1523,
  },
  Object {
    "actualText": "1634",
    "id": "root[&]浙江省[&]舟山市-root[&]办公用品[&]纸张[&]number",
    "value": 1634,
  },
  Object {
    "actualText": "4004",
    "id": "root[&]四川省[&]成都市-root[&]办公用品[&]纸张[&]number",
    "value": 4004,
  },
  Object {
    "actualText": "3077",
    "id": "root[&]四川省[&]绵阳市-root[&]办公用品[&]纸张[&]number",
    "value": 3077,
  },
  Object {
    "actualText": "3551",
    "id": "root[&]四川省[&]南充市-root[&]办公用品[&]纸张[&]number",
    "value": 3551,
  },
  Object {
    "actualText": "352",
    "id": "root[&]四川省[&]乐山市-root[&]办公用品[&]纸张[&]number",
    "value": 352,
  },
]
`;

exports[`Cell Nill Value Tests should get correctly nill cell value 1`] = `
Array [
  Object {
    "actualText": "省份",
    "id": "province",
    "value": "省份",
  },
  Object {
    "actualText": "城市",
    "id": "city",
    "value": "城市",
  },
  Object {
    "actualText": "类别",
    "id": "type",
    "value": "类别",
  },
  Object {
    "actualText": "子类别",
    "id": "sub_type",
    "value": "子类别",
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "浙江省",
    "id": "root[&]浙江省",
    "value": "浙江省",
  },
  Object {
    "actualText": "杭州市",
    "id": "root[&]浙江省[&]杭州市",
    "value": "杭州市",
  },
  Object {
    "actualText": "绍兴市",
    "id": "root[&]浙江省[&]绍兴市",
    "value": "绍兴市",
  },
  Object {
    "actualText": "宁波市",
    "id": "root[&]浙江省[&]宁波市",
    "value": "宁波市",
  },
  Object {
    "actualText": "舟山市",
    "id": "root[&]浙江省[&]舟山市",
    "value": "舟山市",
  },
  Object {
    "actualText": "四川省",
    "id": "root[&]四川省",
    "value": "四川省",
  },
  Object {
    "actualText": "成都市",
    "id": "root[&]四川省[&]成都市",
    "value": "成都市",
  },
  Object {
    "actualText": "绵阳市",
    "id": "root[&]四川省[&]绵阳市",
    "value": "绵阳市",
  },
  Object {
    "actualText": "南充市",
    "id": "root[&]四川省[&]南充市",
    "value": "南充市",
  },
  Object {
    "actualText": "乐山市",
    "id": "root[&]四川省[&]乐山市",
    "value": "乐山市",
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$",
    "value": undefined,
  },
  Object {
    "actualText": "数量",
    "id": "root[&]$$null$$[&]$$null$$[&]number",
    "value": "number",
  },
  Object {
    "actualText": "家具",
    "id": "root[&]家具",
    "value": "家具",
  },
  Object {
    "actualText": "桌子",
    "id": "root[&]家具[&]桌子",
    "value": "桌子",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]家具[&]桌子[&]number",
    "value": "number",
  },
  Object {
    "actualText": "沙发",
    "id": "root[&]家具[&]沙发",
    "value": "沙发",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]家具[&]沙发[&]number",
    "value": "number",
  },
  Object {
    "actualText": "办公用品",
    "id": "root[&]办公用品",
    "value": "办公用品",
  },
  Object {
    "actualText": "笔",
    "id": "root[&]办公用品[&]笔",
    "value": "笔",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]办公用品[&]笔[&]number",
    "value": "number",
  },
  Object {
    "actualText": "纸张",
    "id": "root[&]办公用品[&]纸张",
    "value": "纸张",
  },
  Object {
    "actualText": "数量",
    "id": "root[&]办公用品[&]纸张[&]number",
    "value": "number",
  },
  Object {
    "actualText": "7789",
    "id": "root[&]$$null$$[&]$$null$$-root[&]$$null$$[&]$$null$$[&]number",
    "value": 7789,
  },
  Object {
    "actualText": "-",
    "id": "root[&]浙江省[&]杭州市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]浙江省[&]绍兴市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]浙江省[&]宁波市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]浙江省[&]舟山市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]四川省[&]成都市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]四川省[&]绵阳市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]四川省[&]南充市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]四川省[&]乐山市-root[&]$$null$$[&]$$null$$[&]number",
    "value": null,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$-root[&]家具[&]桌子[&]number",
    "value": null,
  },
  Object {
    "actualText": "7789",
    "id": "root[&]浙江省[&]杭州市-root[&]家具[&]桌子[&]number",
    "value": 7789,
  },
  Object {
    "actualText": "2367",
    "id": "root[&]浙江省[&]绍兴市-root[&]家具[&]桌子[&]number",
    "value": 2367,
  },
  Object {
    "actualText": "3877",
    "id": "root[&]浙江省[&]宁波市-root[&]家具[&]桌子[&]number",
    "value": 3877,
  },
  Object {
    "actualText": "4342",
    "id": "root[&]浙江省[&]舟山市-root[&]家具[&]桌子[&]number",
    "value": 4342,
  },
  Object {
    "actualText": "1723",
    "id": "root[&]四川省[&]成都市-root[&]家具[&]桌子[&]number",
    "value": 1723,
  },
  Object {
    "actualText": "1822",
    "id": "root[&]四川省[&]绵阳市-root[&]家具[&]桌子[&]number",
    "value": 1822,
  },
  Object {
    "actualText": "1943",
    "id": "root[&]四川省[&]南充市-root[&]家具[&]桌子[&]number",
    "value": 1943,
  },
  Object {
    "actualText": "2330",
    "id": "root[&]四川省[&]乐山市-root[&]家具[&]桌子[&]number",
    "value": 2330,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$-root[&]家具[&]沙发[&]number",
    "value": null,
  },
  Object {
    "actualText": "5343",
    "id": "root[&]浙江省[&]杭州市-root[&]家具[&]沙发[&]number",
    "value": 5343,
  },
  Object {
    "actualText": "632",
    "id": "root[&]浙江省[&]绍兴市-root[&]家具[&]沙发[&]number",
    "value": 632,
  },
  Object {
    "actualText": "7234",
    "id": "root[&]浙江省[&]宁波市-root[&]家具[&]沙发[&]number",
    "value": 7234,
  },
  Object {
    "actualText": "834",
    "id": "root[&]浙江省[&]舟山市-root[&]家具[&]沙发[&]number",
    "value": 834,
  },
  Object {
    "actualText": "2451",
    "id": "root[&]四川省[&]成都市-root[&]家具[&]沙发[&]number",
    "value": 2451,
  },
  Object {
    "actualText": "2244",
    "id": "root[&]四川省[&]绵阳市-root[&]家具[&]沙发[&]number",
    "value": 2244,
  },
  Object {
    "actualText": "2333",
    "id": "root[&]四川省[&]南充市-root[&]家具[&]沙发[&]number",
    "value": 2333,
  },
  Object {
    "actualText": "2445",
    "id": "root[&]四川省[&]乐山市-root[&]家具[&]沙发[&]number",
    "value": 2445,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$-root[&]办公用品[&]笔[&]number",
    "value": null,
  },
  Object {
    "actualText": "945",
    "id": "root[&]浙江省[&]杭州市-root[&]办公用品[&]笔[&]number",
    "value": 945,
  },
  Object {
    "actualText": "1304",
    "id": "root[&]浙江省[&]绍兴市-root[&]办公用品[&]笔[&]number",
    "value": 1304,
  },
  Object {
    "actualText": "1145",
    "id": "root[&]浙江省[&]宁波市-root[&]办公用品[&]笔[&]number",
    "value": 1145,
  },
  Object {
    "actualText": "1432",
    "id": "root[&]浙江省[&]舟山市-root[&]办公用品[&]笔[&]number",
    "value": 1432,
  },
  Object {
    "actualText": "2335",
    "id": "root[&]四川省[&]成都市-root[&]办公用品[&]笔[&]number",
    "value": 2335,
  },
  Object {
    "actualText": "245",
    "id": "root[&]四川省[&]绵阳市-root[&]办公用品[&]笔[&]number",
    "value": 245,
  },
  Object {
    "actualText": "2457",
    "id": "root[&]四川省[&]南充市-root[&]办公用品[&]笔[&]number",
    "value": 2457,
  },
  Object {
    "actualText": "2458",
    "id": "root[&]四川省[&]乐山市-root[&]办公用品[&]笔[&]number",
    "value": 2458,
  },
  Object {
    "actualText": "-",
    "id": "root[&]$$null$$[&]$$null$$-root[&]办公用品[&]纸张[&]number",
    "value": null,
  },
  Object {
    "actualText": "1343",
    "id": "root[&]浙江省[&]杭州市-root[&]办公用品[&]纸张[&]number",
    "value": 1343,
  },
  Object {
    "actualText": "1354",
    "id": "root[&]浙江省[&]绍兴市-root[&]办公用品[&]纸张[&]number",
    "value": 1354,
  },
  Object {
    "actualText": "1523",
    "id": "root[&]浙江省[&]宁波市-root[&]办公用品[&]纸张[&]number",
    "value": 1523,
  },
  Object {
    "actualText": "1634",
    "id": "root[&]浙江省[&]舟山市-root[&]办公用品[&]纸张[&]number",
    "value": 1634,
  },
  Object {
    "actualText": "4004",
    "id": "root[&]四川省[&]成都市-root[&]办公用品[&]纸张[&]number",
    "value": 4004,
  },
  Object {
    "actualText": "3077",
    "id": "root[&]四川省[&]绵阳市-root[&]办公用品[&]纸张[&]number",
    "value": 3077,
  },
  Object {
    "actualText": "3551",
    "id": "root[&]四川省[&]南充市-root[&]办公用品[&]纸张[&]number",
    "value": 3551,
  },
  Object {
    "actualText": "352",
    "id": "root[&]四川省[&]乐山市-root[&]办公用品[&]纸张[&]number",
    "value": 352,
  },
]
`;
